components/script/webdriver_handlers.rs

```diff
@@ -398,6 +398,41 @@ pub fn handle_find_element_elements_css(
     reply.send(node_ids).unwrap();
 }
 
+pub fn handle_find_element_elements_link_text(
+    documents: &Documents,
+    pipeline: PipelineId,
+    element_id: String,
+    selector: String,
+    partial: bool,
+    reply: IpcSender<Result<Vec<String>, ()>>,
+) {
+    let node_ids = find_node_by_unique_id(documents, pipeline, element_id)
+        .ok_or(())
+        .and_then(|node| {
+            node.query_selector_all(DOMString::from("a"))
+                .map_err(|_| ())
+        })
+        .map(|nodes| {
+            nodes
```

Comment:
To reduce a lot of the duplication in these commits, could we refactor the code so we have:
* a `all_matching_links` helper that accepts a root node, performs the query_selector_all operation, and returns a filtered iterator of unique ids
* a `first_matching_link` helper that calls `all_matching_links` and uses `take(1)` to return the first element of the filtered iterator

Using these building blocks, we should be able to write much more concise and less repetitive implementations for all of the new webdriver handlers.

Reply:
I tried to return an iterator from `all_matching_links` but kept getting `Returns a value referencing data owned by the current function` compile-time errors. Instead, I simply collected the iterator and returned a vector of ids. Any thoughts on how to make it return an iterator?